### PR TITLE
TiFlash: Modify scale doc structure

### PR DESCRIPTION
**Pull request by birdstorm**
### What is changed, added or deleted? (Required)

TiUP scale-in command has supported to delete TiFlash store from PD. Thus some modification will be made for the documents.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

**Comment by yikeke** [comment]
This PR may conflict with https://github.com/pingcap/docs-cn/pull/3208/. Do you @birdstorm and @King-Dylan commit different changes to `scale-tidb-using-tiup.md`?

**Comment by birdstorm** [comment]
> This PR may conflict with #3208. Do you @birdstorm and @King-Dylan commit different changes to `scale-tidb-using-tiup.md`?

I will modify my PR after #3208 is merged.

**Comment by yikeke** [comment]
> > This PR may conflict with #3208. Do you @birdstorm and @King-Dylan commit different changes to `scale-tidb-using-tiup.md`?
> 
> I will modify my PR after #3208 is merged.

ok

**Comment by yikeke** [comment]
> #3208

[#3208](https://github.com/pingcap/docs-cn/pull/3208) is merged. Please resolve conflicts. @birdstorm

**Comment by yikeke** [comment]
> 扩容的文档还没合，等先把这个合并了 #3208

[#3208](https://github.com/pingcap/docs-cn/pull/3208) is merged. Please resolve conflicts. @birdstorm

**Comment by birdstorm** [comment]
@yikeke @kissmydb  PTAL, thx

File: scale-tidb-using-tiup.md [hunk 1]
```diff
@@ -309,27 +309,114 @@ tiup cluster display <cluster-name>
...
+2. 调用 scale-in 命令执行下线节点，例如步骤 1 中获得该节点名为 `10.0.1.4:9000`
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    tiup cluster scale-in <cluster-name> --node 10.0.1.4:9000
+    ```
+
+### 5.2 手动缩容 TiFlash 节点
+
```
Comment (flowbehappy) [inline scale-tidb-using-tiup.md:345]:
```suggestion
在特殊情况下，比如需要强制下线节点，或者 TiUP 操作失败的情况下，可以使用以下方法手动下线 TiFlash 节点。
```

File: scale-tidb-using-tiup.md [hunk 2]
```diff
@@ -309,27 +309,114 @@ tiup cluster display <cluster-name>
...
+1. 使用 pd-ctl 的 store 命令在 PD 中查看该 TiFlash 节点对应的 store id。
+
+    * 在 [pd-ctl](/pd-control.md) (tidb-ansible 目录下的 `resources/bin` 包含对应的二进制文件) 中输入 store 命令。
+
+    * 若使用 TiUP 部署，则请调用以下命令：
```
Comment (flowbehappy) [inline scale-tidb-using-tiup.md:350]:
```suggestion
    * 若使用 TiUP 部署，可以调用以下命令代替：
```

File: scale-tidb-using-tiup.md [hunk 3]
```diff
@@ -309,27 +309,114 @@ tiup cluster display <cluster-name>
...
+
+      ```shell
+      tiup ctl pd -u <pd-address> store
+      ```
+
+2. 在 pd-ctl 中下线该 TiFlash 节点。
+
+    * 在 pd-ctl 中输入 `store delete <store_id>`，其中 <store_id> 为上一步查到的该 TiFlash 节点对应的 store id。
+
+    * 若通过 TiUP 部署，则请调用以下命令：
```
Comment (flowbehappy) [inline scale-tidb-using-tiup.md:362]:
```suggestion
    * 若通过 TiUP 部署，可以调用以下命令代替：
```

File: scale-tidb-using-tiup.md [hunk 4]
```diff
@@ -309,27 +309,114 @@ tiup cluster display <cluster-name>
...
+
+      {{< copyable "shell-regular" >}}
+
+      ```shell
+      tiup ctl pd -u <pd-address> store delete <store_id>
+      ```
+
+3. 等待该 TiFlash 节点对应的 store 消失或者 state_name 变成 Tombstone 再关闭 TiFlash 进程。
+
+4. 手动删除 TiFlash 的数据文件，其目录由用户先前配置于 TiFlash 配置文件中。
```
Comment (flowbehappy) [inline scale-tidb-using-tiup.md:372]:
```suggestion
4. 手动删除 TiFlash 的数据文件，具体位置请查看 TiFlash 的配置文件。
```

File: scale-tidb-using-tiup.md [hunk 5]
```diff
@@ -309,27 +309,114 @@ tiup cluster display <cluster-name>
 
 > **注意：**
 >
-> 本节介绍的缩容流程不会删除缩容节点上的数据文件，如需再次上线该节点，请先手动删除原来的数据文件。
+> 如果下线该 TiFlash 节点后，TiFlash 集群剩余节点数大于等于所有数据表的最大副本数，可跳过下列步骤。
```
Comment (flowbehappy) [inline scale-tidb-using-tiup.md:312]: 建议这里改成一个具体的步骤，即 5.1.

```suggestion
### 5.1 在下线节点之前，确保 TiFlash 集群剩余节点数大于等于所有数据表的最大副本数，否则需要修改相关表的 TiFlash 副本数。
```

**Comment by birdstorm** [comment]
@yikeke @kissmydb PTAL again, thx

File: scale-tidb-using-tiup.md [hunk 6]
```diff
@@ -307,29 +307,118 @@ tiup cluster display <cluster-name>
...
+
+    * 若通过 TiUP 部署，可以调用以下命令代替：
+
+      {{< copyable "shell-regular" >}}
+
+      ```shell
+      tiup ctl pd -u <pd-address> store delete <store_id>
+      ```
+
+3. 等待该 TiFlash 节点对应的 store 消失或者 state_name 变成 Tombstone 再关闭 TiFlash 进程。
```
Comment (flowbehappy) [inline scale-tidb-using-tiup.md:372]: 可以加上一句：

如果等待较长时间后，该节点仍然无法正常消失或者状态变成 Tombstone，可以考虑以下命令，把节点强制踢出集群：
**注意以下命令会直接丢弃该 TiFlash 节点上的副本，有可能导致查询失败**
```
curl -X POST 'http://pd:2379/pd/api/v1/store/{id}/state?state=Tombstone'
```

**Comment by flowbehappy** [comment]
LGTM

File: scale-tidb-using-tiup.md [hunk 7]
```diff
@@ -307,29 +307,128 @@ tiup cluster display <cluster-name>
...
+
+### 5.3 手动缩容 TiFlash 节点
+
+在特殊情况下，比如需要强制下线节点，或者 TiUP 操作失败的情况下，可以使用以下方法手动下线 TiFlash 节点。
+
+1. 使用 pd-ctl 的 store 命令在 PD 中查看该 TiFlash 节点对应的 store id。
+
+    * 在 [pd-ctl](/pd-control.md) (tidb-ansible 目录下的 `resources/bin` 包含对应的二进制文件) 中输入 store 命令。
+
+    * 若使用 TiUP 部署，可以调用以下命令代替：
```
Comment (kissmydb) [inline scale-tidb-using-tiup.md:352]:
```suggestion
    * 若使用 TiUP 部署，可以调用以下命令代替 `pd-ctl`：
```

File: scale-tidb-using-tiup.md [hunk 8]
```diff
@@ -307,29 +307,128 @@ tiup cluster display <cluster-name>
...
+
+      ```shell
+      tiup ctl pd -u <pd-address> store
+      ```
+
+2. 在 pd-ctl 中下线该 TiFlash 节点。
+
+    * 在 pd-ctl 中输入 `store delete <store_id>`，其中 <store_id> 为上一步查到的该 TiFlash 节点对应的 store id。
+
+    * 若通过 TiUP 部署，可以调用以下命令代替：
```
Comment (kissmydb) [inline scale-tidb-using-tiup.md:364]:
```suggestion
    * 若通过 TiUP 部署，可以调用以下命令代替 `pd-ctl`：
```

File: scale-tidb-using-tiup.md [hunk 9]
```diff
@@ -307,29 +307,128 @@ tiup cluster display <cluster-name>
 
 如果要缩容 IP 地址为 10.0.1.4 的一个 TiFlash 节点，，可以按照如下步骤进行操作。
```
Comment (yikeke) [inline scale-tidb-using-tiup.md:308]:
```suggestion
如果要缩容 IP 地址为 10.0.1.4 的一个 TiFlash 节点，可以按照如下步骤进行操作。
```

File: scale-tidb-using-tiup.md [hunk 10]
```diff
@@ -307,29 +307,128 @@ tiup cluster display <cluster-name>
...
-```
+1. 通过以下命令确定需要下线的节点名称：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    tiup cluster display <cluster-name>
+    ```
+
+2. 调用 scale-in 命令执行下线节点，例如步骤 1 中获得该节点名为 `10.0.1.4:9000`
```
Comment (yikeke) [inline scale-tidb-using-tiup.md:336]:
```suggestion
2. 执行 scale-in 命令来下线节点，假设步骤 1 中获得该节点名为 `10.0.1.4:9000`
```

File: scale-tidb-using-tiup.md [hunk 11]
```diff
@@ -307,29 +307,128 @@ tiup cluster display <cluster-name>
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    tiup cluster scale-in <cluster-name> --node 10.0.1.4:9000
+    ```
+
+### 5.3 手动缩容 TiFlash 节点
+
+在特殊情况下，比如需要强制下线节点，或者 TiUP 操作失败的情况下，可以使用以下方法手动下线 TiFlash 节点。
```
Comment (yikeke) [inline scale-tidb-using-tiup.md:346]:
```suggestion
在特殊情况下（比如需要强制下线节点），或者 TiUP 操作失败的情况下，可以使用以下方法手动下线 TiFlash 节点。
```

File: scale-tidb-using-tiup.md [hunk 12]
```diff
@@ -307,29 +307,128 @@ tiup cluster display <cluster-name>
...
+    * 在 [pd-ctl](/pd-control.md) (tidb-ansible 目录下的 `resources/bin` 包含对应的二进制文件) 中输入 store 命令。
+
+    * 若使用 TiUP 部署，可以调用以下命令代替 `pd-ctl`：
 
-### 5.4 查看集群状态
+      {{< copyable "shell-regular" >}}
+
+      ```shell
+      tiup ctl pd -u <pd-address> store
+      ```
```
Comment (yikeke) [inline scale-tidb-using-tiup.md:358]: 这里整体需要再缩进四格

File: scale-tidb-using-tiup.md [hunk 13]
```diff
@@ -307,29 +307,128 @@ tiup cluster display <cluster-name>
...
+
+    * 在 pd-ctl 中输入 `store delete <store_id>`，其中 <store_id> 为上一步查到的该 TiFlash 节点对应的 store id。
+
+    * 若通过 TiUP 部署，可以调用以下命令代替 `pd-ctl`：
+
+      {{< copyable "shell-regular" >}}
+
+      ```shell
+      tiup ctl pd -u <pd-address> store delete <store_id>
+      ```
```
Comment (yikeke) [inline scale-tidb-using-tiup.md:370]: 这里整体需要再缩进四格

File: upgrade-tidb-using-tiup.md [hunk 14]
```diff
@@ -198,6 +198,10 @@ tiup cluster upgrade <cluster-name> v4.0.0-rc --force
 https://download.pingcap.org/tidb-{version}-linux-amd64.tar.gz
 ```
 
+### 5.4 升级过程中 TiFlash 组件升级失败
+
+由于 TiFlash 在 `v4.0.0-rc.2` 之前的版本可能有一些不兼容的问题，所以如果将包含 TiFlash 组件的集群由此前版本升级至 `v4.0.0-rc.2` 之后的过程中遇到问题，请访问 [ASK TUG](https://asktug.com/) 提出，会由研发方面人工解决。
```
Comment (yikeke) [inline upgrade-tidb-using-tiup.md:203]:
```suggestion
TiFlash 在 `v4.0.0-rc.2` 之前的版本可能有一些不兼容的问题。因此，如果将包含 TiFlash 组件的集群由此前版本升级至 `v4.0.0-rc.2` 之后版本的过程中遇到问题，可在 [ASK TUG](https://asktug.com/) 反馈，寻求研发人员支持。
```

**Comment by yikeke** [comment]
我们现在统一不在标题里加编号了，我删一下，稍等再合

**Comment by birdstorm** [comment]
@yikeke Sure. Just one question,  how about `upgrade-tidb-using-tiup.md`?

**Comment by yikeke** [comment]
> @yikeke Sure. Just one question, how about `upgrade-tidb-using-tiup.md`?
> 
> edit: Never mind, it does not bother with this PR.

Let's modify its format later. We have 30 or more PRs to be merged today. First thing first. If you have any open PR, you can modify it in your PR.

**Pull request merged**